### PR TITLE
Helper functions for the repeater fields

### PR DESCRIPTION
**Pull request by lukecarbis**
This PR adds the following helper functions:

- `block_rows()`
- `block_row()`
- `block_sub_field()`
- `block_sub_value()`

### CODE REVIEW

**Comment by lukecarbis** [comment]
@kienstra Ready for you to review.

File: php/blocks/class-loop.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	 * @var string
+	 */
+	public $active;
+
+	/**
+	 * Set a loop to active.
+	 *
+	 * @param string $name The field name.
+	 */
+	public function active( $name ) {
```
Comment (kienstra) [inline php/blocks/class-loop.php]: OK, nice. It looks like only one row can be active at a time, though there might be a few rows stored in `$this->loops`.

File: php/blocks/class-loop.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * @copyright Copyright(c) 2019, Block Lab
+ * @license http://opensource.org/licenses/GPL-2.0 GNU General Public License, version 2 (GPL-2.0)
+ */
+
+namespace Block_Lab\Blocks;
+
+/**
+ * Class Loop
+ */
+class Loop {
```
Comment (kienstra) [inline php/blocks/class-loop.php:15]: It's nice how this clas follows the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) really well. It only handles the loop, and doesn't have to know about what it's looping through.

**Review by kienstra — APPROVED** [review]
**Approved**

Hi @lukecarbis,
Good work! This looks good, and will help a lot with the progress on the `feature/repeater` branch.

**Comment by kienstra** [comment]
For reference, here's how to use these repeater helper functions in a template:

```php
$repeater_name = 'my-repeater';
if ( block_rows( $repeater_name ) ) :
	while ( block_rows( $repeater_name ) ) :
		block_row( $repeater_name );
		
		block_sub_field( 'your-example-field' );
	endwhile;
endif;

```

**Pull request merged**